### ISSUE TITLE
Failing to set the TrustedCerts gives useless error message 

### ISSUE THREAD

**Issue by ericwol-msft** [Azure/azure-iot-sdk-c#1108]
**Repro:**
When trying to conned the Azure Sphere directly to an IoT hub (not using DPS) you need to set the TrustedCerts option of the root cert (Baltimore CyberTrust Root)
If you fail to do this, you get an IOTHUB_CLIENT_CONNECTION_STATUS with a reason of IOTHUB_CLIENT_CONNECTION_NO_NETWORK

**Issue:**
Failing SSL connection gives an result of IOTHUB_CLIENT_CONNECTION_UNAUTHENTICATED and a reason code of IOTHUB_CLIENT_CONNECTION_NO_NETWORK.

**Expected:**
The reason code should be of value that indicates that connection could not be made because of a SSL issue. I spent a lot of time trying to figure out why my Sphere was not connected to the wireless network due to the above error code.

**Comment by danewalton** [comment]
You are right. As it stands the handshake is somewhat black boxed in that we don't have specific details into which part of the handshake failed from the sdk perspective. We will look into adding a log error message to make this easier to debug.

**Comment by danewalton** [comment]
Hi Eric I went ahead and repro'd this situation using wolfssl and I was able to figure out what the probable cause of the runtime error might be if I was in this situation by turning on the trace option. 

```c
bool traceOn = true;
IoTHubDeviceClient_LL_SetOption(device_ll_handle, OPTION_LOG_TRACE, &traceOn);
```

Using this I received the following error message:
`Error: Time:Wed Aug 21 22:21:11 2019 File:/home/repos/azure-iot-sdk-c/c-utility/adapters/tlsio_wolfssl.c Func:tlsio_wolfssl_open Line:752 Failed to connect to server.  The certificates may not be correct.`
Due to the blackboxing of the tls adapter layer we currently can't give a more in depth reason for the failure of the tls handshake but that can be rectified using the logs. Hopefully that helps!
